File: civitasRuby/pkg/civitasRuby-0.0.1/lib/civitas/sorpresa_especulador.rb
#encoding:utf-8

=begin
Authors: Esther García Gallego
         Yesenia González Dávila
         Grupo B3
=end

require_relative 'jugador_especulador'

module Civitas

class SorpresaEspeculador < Sorpresa
  
  def initialize (fianza)
    super("*Especulador*", nil, fianza, "Ahora eres un jugador especulador.", nil)
  end
  
  def aplicarAJugador (actual, todos)
    if jugadorCorrecto(actual, todos)
      informe(actual, todos)
      jugadorEspeculador = JugadorEspeculador.nuevoEspeculador(todos[actual], @valor)
      todos[actual] = jugadorEspeculador
    end
  end
  
end

end

File: civitasJava/src/civitas/SorpresaEspeculador.java

package civitas;

import java.util.ArrayList;


public class SorpresaEspeculador extends Sorpresa {
    
    private float fianza;
    
    SorpresaEspeculador (float fianza){
        super ("*Especulador*");
        this.fianza = fianza;
    }
    
    /** Se cambia al jugador por un jugador especulador.  */
    @Override
    void aplicarAJugador (int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe (actual, todos);
            Jugador especulador = new JugadorEspeculador (todos.get(actual), fianza);
            todos.set(actual, especulador);
        }
    }
    
}
